quic/s2n-quic-transport/src/space/application.rs

```diff
@@ -166,7 +166,7 @@ impl<Config: endpoint::Config> ApplicationSpace<Config> {
         // necessary) and record only if a packet is transmitted.
         let mut skipped_packet_number = SkippedPacketNumber::default();
 
-        if self.recovery_manager.requires_probe() {
+        if self.recovery_manager.requires_probe() && packet_number.as_u64() != 0 {
```

Comment:
are we missing the `PacketSkipped` event here and in 180-198 too? or is it getting emitted somewhere else?

Reply:
Looks like the PacketSkipped event actually gets emitted in the on_packet_sent function on line 272. Seems like we didn't want to [record a packet number as skipped if the packet wasn't actually sent](https://github.com/aws/s2n-quic/blob/34f5e4b054d0fb07c767f7fe63d3c50701df4f6d/quic/s2n-quic-transport/src/space/application.rs#L164).

Reply:
This question did make me consider, it does look like there's actually two reasons a packet number could be skipped in the application space, pto probing and optimistic ack protection. I wondered if our optimistic ack protection could cause packet number 0 to be skipped. However, [looking at the Optimistic Ack code](https://github.com/aws/s2n-quic/blob/34f5e4b054d0fb07c767f7fe63d3c50701df4f6d/quic/s2n-quic-transport/src/space/application.rs#L528) I think that the minimum skip counter is 30, so, we would never skip pn 0.